src/syntax.ml

```diff
@@ -86,6 +86,8 @@ type p_tactic_aux =
   (** Print the current goal. *)
   | P_tac_proofterm
   (** Print the current proof term (possibly containing open goals). *)
+  | P_tac_assert of (ident option * p_term)
```

Comment:
I'm guessing you don't want parentheses here. The memory representation is actually not the same with and without parentheses. Without them, both elements of the product are stored in the constructor, but with parentheses the constructor only stores a pointer to a couple (thus there is an extra indirection).

Note also that the order of the two arguments should be reversed to match the syntax (the as comes second).